interface/patient_file/pos_checkout_normal.php

```diff
@@ -0,0 +1,1158 @@
+{
+    $amount = sprintf('%01.2f', 0 - $amount); // make it negative
+    echo " <tr>\n";
+    echo "  <td>" . text(oeFormatShortDate($paydate)) . "</td>\n";
+    echo "  <td>" . xlt('Payment') . " " . text($description) . "</td>\n";
+    echo "  <td colspan='2'>&nbsp;</td>\n";
+    echo "  <td class='text-right'>" . text(oeFormatMoney($amount)) . "</td>\n";
+    echo " </tr>\n";
+}
+
+// Generate a receipt from the last-billed invoice for this patient,
+// or for the encounter specified as a GET parameter.
+//
+function generate_receipt($patient_id, $encounter = 0)
+{
+ //REMEMBER the entire receipt is generated here, have to echo DOC type etc and closing tags to create a valid webpsge
+    global $sl_err, $sl_cash_acc, $details, $facilityService;
+
+    // Get details for what we guess is the primary facility.
+    $frow = $facilityService->getPrimaryBusinessEntity(array("useLegacyImplementation" => true));
+
+    $patdata = getPatientData($patient_id, 'fname,mname,lname,pubpid,street,city,state,postal_code,providerID');
+
+    // Get the most recent invoice data or that for the specified encounter.
+    //
+    // Adding a provider check so that their info can be displayed on receipts
+    if ($encounter) {
+        $ferow = sqlQuery("SELECT id, date, encounter, provider_id FROM form_encounter " .
+        "WHERE pid = ? AND encounter = ?", array($patient_id,$encounter));
+    } else {
+        $ferow = sqlQuery("SELECT id, date, encounter, provider_id FROM form_encounter " .
+        "WHERE pid = ? " .
+        "ORDER BY id DESC LIMIT 1", array($patient_id));
+    }
+    if (empty($ferow)) {
+        die(xlt("This patient has no activity."));
+    }
+    $trans_id = $ferow['id'];
+    $encounter = $ferow['encounter'];
+    $svcdate = substr($ferow['date'], 0, 10);
+
+    if ($GLOBALS['receipts_by_provider']) {
+        if (isset($ferow['provider_id'])) {
+            $encprovider = $ferow['provider_id'];
+        } elseif (isset($patdata['providerID'])) {
+            $encprovider = $patdata['providerID'];
+        } else {
+            $encprovider = -1;
+        }
+    }
+
+    if ($encprovider) {
+        $providerrow = sqlQuery("SELECT fname, mname, lname, title, street, streetb, " .
+        "city, state, zip, phone, fax FROM users WHERE id = ?", array($encprovider));
+    }
+
+    // Get invoice reference number.
+    $encrow = sqlQuery("SELECT invoice_refno FROM form_encounter WHERE " .
+    "pid = ? AND encounter = ? LIMIT 1", array($patient_id,$encounter));
+    $invoice_refno = $encrow['invoice_refno'];
+    ?>
+    <!-- being deliberately echoed to indicate it is part of the php function generate_receipt -->
+
+    <!DOCTYPE html>
+    <html>
+    <head>
+        <?php Header::setupHeader(['datetime-picker']);?>
+        <title><?php echo xlt('Receipt for Payment'); ?></title>
+        <script>
+
+        <?php require($GLOBALS['srcdir'] . "/restoreSession.php"); ?>
```

Comment:
@sunsetsystems , This type of thing has always confused me. Shouldn't this script be only in `top` since even popups look for it in `top`?

Reply:
Yes. Long ago I put this in dialogs but later realized that was unnecessary.